node/src/components/fetcher.rs

```diff
@@ -467,14 +476,26 @@ impl ItemFetcher<BlockWithMetadata> for Fetcher<BlockWithMetadata> {
         peer: NodeId,
         responder: FetchResponder<BlockWithMetadata>,
     ) -> Effects<Event<BlockWithMetadata>> {
-        effect_builder
-            .get_block_and_sufficient_finality_signatures_by_height_from_storage(id)
-            .event(move |result| Event::GetFromStorageResult {
-                id,
-                peer,
-                maybe_item: Box::new(result),
-                responder,
-            })
+        let finality_threshold_fraction = self.finality_threshold_fraction;
+        async move {
+            let block_with_metadata = effect_builder
+                .get_block_with_metadata_from_storage_by_height(id, false)
+                .await?;
+            has_enough_block_signatures(
+                effect_builder,
+                block_with_metadata.block.header(),
+                &block_with_metadata.finality_signatures,
```

Comment:
Maybe this field should also be renamed.

```suggestion
                &block_with_metadata.block_signatures,
```